### PR TITLE
Update Google auth approach to use new libraries

### PR DESCRIPTION
**Pull request by daikema**
**NOTE: This is a breaking change for those using Google auth and will require updates to the controller config and Google auth registration as well as a controller install with the code from the [corresponding controller pull request](https://github.com/CARTAvis/carta-controller/pull/166) to work.**

**Description**

The vast majority of this work was on the controller side, and described in the other related pull request.

As far as the new approach goes, following initial login with Google credentials sessions are persisted using access tokens + refresh tokens generated like those for the other auth methods.  As a result, this is mostly removing existing Google code and libraries no longer needed in the frontend.

### CODE REVIEW

File: src/services/ApiService.ts [hunk 1]
```diff
@@ -11,9 +11,9 @@ const snippetSchema = require("models/snippet_schema_1.json");
 export interface RuntimeConfig {
     dashboardAddress?: string;
     apiAddress?: string;
-    googleClientId?: string;
     tokenRefreshAddress?: string;
     logoutAddress?: string;
+    logoutUsingGet?: boolean;
```
Comment (veggiesaurus) [inline src/services/ApiService.ts]: shouldn't this rather be `logoutUsingRedirect`? since we're redirecting the current page to that URL.

**Review by veggiesaurus — CHANGES_REQUESTED** [review]

File: src/services/ApiService.ts (hunk 1 above)
Comment (daikema) [reply src/services/ApiService.ts]: As per Slack discussion now updated code to always use the redirect approach to perform a logout.

**Review by YuHsuan-Hwang — APPROVED** [review]
Looks good.

**Comment by kswang1029** [comment]
@daikema Can this PR be paired with the dev branch of the backend without involving the controller part? Based on what I see here is a no. Therefore, no e2e tests can pass...

This is how I start the backend

```
./carta_backend --frontend_folder /Users/kswang/carta_build/carta-frontend/build --port 3002 --omp_threads 8 --debug_no_auth --no_browser --verbosity 5

[2024-03-14 03:00:26.432Z] [CARTA] [info] Writing to the log file: /Users/kswang/.carta/log/carta.log
[2024-03-14 03:00:26.432Z] [CARTA] [debug] Using command-line settings
[2024-03-14 03:00:26.432Z] [CARTA] [info] /Users/kswang/carta_build/carta-backend/build/carta_backend: Version 5.0.0-dev
[2024-03-14 03:00:26.432Z] [CARTA] [info] Serving CARTA frontend from /Users/kswang/carta_build/carta-frontend/build
[2024-03-14 03:00:26.433Z] [CARTA] [info] Listening on port 3002 with top level folder /, starting folder /Users/kswang/carta_build/carta-backend/build, and 8 OpenMP worker threads
[2024-03-14 03:00:26.433Z] [CARTA] [info] CARTA is accessible at http://localhost:3002
[2024-03-14 03:00:26.433Z] [CARTA] [debug] The CARTA database API is accessible at http://localhost:3002/api/database/...

```

and with http://localhost:3002/ I see something like "cannot authenticate with server".

**Comment by daikema** [comment]
@kswang1029 when you say: 

> with http://localhost:3002/ I see something like "cannot authenticate with server".

Am I correct that this is a controller-deploy without reverse proxy in front?  Is what precisely is showing this error?  Is this the browser displaying a page?  In the debug console?  The backend trying to access controller settings?

**Comment by kswang1029** [comment]
> @kswang1029 when you say: 
> 
> 
> 
> > with http://localhost:3002/ I see something like "cannot authenticate with server".
> 
> 
> 
> Am I correct that this is a controller-deploy without reverse proxy in front?  Is what precisely is showing this error?  Is this the browser displaying a page?  In the debug console?  The backend trying to access controller settings?

No it is just the frontend + backend.  This is what I saw:

https://github.com/CARTAvis/carta-frontend/assets/20819712/6410e592-fff5-47fd-9d17-4f43ac053a2d

**Comment by daikema** [comment]
@kswang1029 can you let me know if the latest changes fix the issues you were seeing?  I think the issue was related to your use of `--debug_no_auth` which these latest changes adjust to handle.

**Comment by kswang1029** [comment]
> @kswang1029 can you let me know if the latest changes fix the issues you were seeing? I think the issue was related to your use of `--debug_no_auth` which these latest changes adjust to handle.

Yes, now it works. I have run e2e tests and do not see regressions. 👍 

@acdo2002 could you please set up a controller instance at ASIAA for testing purpose?

**Comment by acdo2002** [comment]
@kswang1029 Yes, I am trying to do it. but this is my first to work with controller I need time to testing it then apply to the asiaa server, I will let you know when I start to set up a testing instance on the ASIAA server.

**Review by kswang1029 — APPROVED** [review]

**Review by acdo2002 — APPROVED** [review]
ASIAA set up this new carta-frontend PR (+ carta-controller PR) and it works fine.

**Review by veggiesaurus — APPROVED** [review]

**Pull request merged**